fe/fe-core/src/test/java/org/apache/doris/nereids/rules/expression/check/CheckCastTest.java

```diff
@@ -0,0 +1,245 @@
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+
+        // 2. Null type.
+        originType = NullType.INSTANCE;
+        Assertions.assertTrue(CheckCast.check(originType, targetType));
+
+        // 3. Same type.
+        originType = MapType.SYSTEM_DEFAULT;
+        Assertions.assertTrue(CheckCast.check(originType, targetType));
+
+        // 4. ipv4 can't cast to boolean, numeric, date like and time
+        checkCastToBasicType(IPv4Type.INSTANCE);
+
+        // 5. ipv6 can't cast to boolean, numeric, date like and time
+        checkCastToBasicType(IPv6Type.INSTANCE);
+
+        // 6. bitmap can't cast to boolean, numeric, date like and time
+        checkCastToBasicType(BitmapType.INSTANCE);
+        checkCastToIpType(BitmapType.INSTANCE);
+
+        // 7. bitmap can't cast to boolean, numeric, date like and time
+        checkCastToBasicType(HllType.INSTANCE);
+        checkCastToIpType(HllType.INSTANCE);
+
+        // 8. array can't cast to boolean, numeric, date like and time
+        checkCastToBasicType(ArrayType.SYSTEM_DEFAULT);
+        checkCastToIpType(ArrayType.SYSTEM_DEFAULT);
+
+        // 9. array can't cast to boolean, numeric, date like and time
+        checkCastToBasicType(MapType.SYSTEM_DEFAULT);
+        checkCastToIpType(MapType.SYSTEM_DEFAULT);
+
+        // 10. array can't cast to boolean, numeric, date like and time
+        checkCastToBasicType(StructType.SYSTEM_DEFAULT);
+        checkCastToIpType(StructType.SYSTEM_DEFAULT);
+
+        // 11. boolean can't cast to date like and time
+        originType = BooleanType.INSTANCE;
+        targetType = DateType.INSTANCE;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        targetType = DateTimeV2Type.MAX;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        targetType = TimeV2Type.MAX;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        checkCastToIpType(BooleanType.INSTANCE);
+
+        // 12. date like, time and ip can't cast to boolean
+        originType = DateType.INSTANCE;
+        targetType = BooleanType.INSTANCE;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        originType = DateTimeV2Type.MAX;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        originType = TimeV2Type.MAX;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        originType = IPv4Type.INSTANCE;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        originType = IPv6Type.INSTANCE;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+    }
+
+    private void checkCastToBasicType(DataType originType) {
+        DataType targetType = BooleanType.INSTANCE;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
+        targetType = TinyIntType.INSTANCE;
+        Assertions.assertFalse(CheckCast.check(originType, targetType));
```

Comment:
It would be clearer not to use local variables
```suggestion
        Assertions.assertFalse(CheckCast.check(originType, TinyIntType.INSTANCE));
```